docs/diagrams/wallet-lifecycle/wallet-lifecycle-4-resolving-moving-funds.tikz

```diff
@@ -0,0 +1,121 @@
+% needed packages (recommended, downloads only essential packages):
+% ```
+% brew install --cask basictex
+% sudo tlmgr update --self
+% sudo tlmgr install standalone
+% sudo tlmgr install comment
+% pdflatex <filename>.tikz
+%```
+% 
+% To generate `.png` file out of `.pdf` on MacOS run:
+% `convert -density 300 <filename>.pdf -resize 50% <filename>.png`.
+% This requires installation of the `imagemagick` tool:
+% `brew install imagemagick`.
+\documentclass{standalone}%
+\usepackage{tikz}%
+\usetikzlibrary{positioning,shapes,arrows.meta,chains}%
+
+% These commands are used to draw arrow labels
+\newcommand{\yes}{node [auto, pos=0, xshift=5mm]{Y}}%
+\newcommand{\no}{node [auto, swap, pos=0, xshift=-5mm]{N}}%
+
+% We need layers to draw the background
+\pgfdeclarelayer{background}%
+\pgfsetlayers{background,main}%
+
+\begin{document}%
+\begin{tikzpicture}[
+% Specify styles for nodes:
+    base/.style = {rectangle, rounded corners, draw=black,
+                   minimum width=4cm, minimum height=1cm,
+                   text centered, text width=4cm, align=flush center},
+   start/.style = {base, fill=blue!30},
+  revert/.style = {base, fill=red!30},
+  action/.style = {base, fill=black!5},
+   check/.style = {diamond, aspect=1.2, draw=black, fill=black!5,
+                   minimum width=1cm, align=center},
+   note1/.style = {rectangle, draw, dashed, fill=green!10,
+                   align=flush left, text width=5cm},
+   note2/.style = {rectangle, draw, dashed, fill=cyan!10,
+                   align=flush left, text width=5cm},
+              > = Triangle,                  % defines the arrow tip style when `->` is used
+           font = \sffamily,
+]
+
+% MOVING FUNDS PROOF
+% Nodes
+\node[start]                                                      (mfp_start)   {Somebody sends \textbf{proof of moving funds}};
+\node[check, below=of mfp_start]                                  (mfp_check1)  {Did wallet send\\ a commitment to\\ target wallets?};
+\node[check, below right=of mfp_check1]                           (mfp_check2)  {Has the\\ transaction\\ occurred on\\ Bitcoin?};
+\node[check, below right=of mfp_check2]                           (mfp_check3)  {Does the\\ transaction have\\ a main UTXO in\\ its inputs?};
+\node[check, below right=of mfp_check3]                           (mfp_check4)  {Is wallet\\ MOVING\_FUNDS?};
+\node[check, below right=of mfp_check4]                           (mfp_check5)  {Are the\\ outputs of equal value\\ ($\pm$ reminder from dividing\\ sum\_of\_outputs /\\ no\_of\_outputs)?};
+\node[check, below right=of mfp_check5]                           (mfp_check6)  {Were outputs\\ sent to wallets\ declared in the\\ target\_wallet\_commitment?};
+\node[action, below right=of mfp_check6]                          (mfp_action1) {Register moved funds sweep requests for the source wallets};
+\node[check, below=of mfp_action1]                                (mfp_check7)  {Isn't the fee\\ too high?};         
```

Comment:
*the Bitcoin fee

Reply:
Also, can we put

> Isn't the Bitcoin fee too high?

BEFORE

> Register moved funds sweep requests for the target wallets?

It is not exactly how it's organized in the code, but I think this way is much easier to reason about.